src/rideables/MontageMSQueue.hpp

```diff
@@ -29,20 +28,23 @@ class MontageMSQueue : public RQueue<T>{
 
 private:
     struct Node{
-        atomic_nbptr_t<Node*> next;
+        MontageMSQueue* ds;
```

Comment:
(Applicable to all data structures) I suggest to make this a reference (e.g., MontageMSQueue&), to make code more cpp-ish (using less pointers).

Reply:
That said, storing pointer/reference to a parent instance in all nodes is still awkward. I don't have a better idea for now though..

Reply:
The issue here is to let destructor know which data structure is responsible for the reclamation of payload(s). We kind of have to do this as destructors don't take arguments.

Reply:
Yes I agree with this.